input/taproot.go

```diff
@@ -43,15 +43,15 @@ func MultiPrevOutFetcher(inputs []Input) (*txscript.MultiPrevOutFetcher, error)
 		op := inp.OutPoint()
 		desc := inp.SignDesc()
 
-		if op == nil {
+		if op == EmptyOutPoint {
```

Comment:
K now I see what's going on. This also seems sus to me. When is it possible for `op` to be the 0-valued outpoint.

Reply:
Not sure - this is needed to have the same effect as `op == nil` check, also I remember we do use zero chan ID or zero SCID to signal we wanna do something to all of our channels, think it's in the specs? cc @Crypt-iQ

Reply:
zero chan id is to send an error / warning to all chans

Reply:
Yeah it looks actually like the all-zero chan ID has different semantics depending on if it's a [`warning` or an `error`](https://github.com/lightning/bolts/blob/master/01-messaging.md#requirements-2)

I also want to note that while ChannelIDs are deterministic projections of outpoints, they are not the same conceptual objects and should not be treated as interchangeable.

I'd like to encourage us to code in a way where we write our logic in the semantic domain of the objects themselves rather than the semantic domain of their representation.

The central question here is: 

> Does the `EmptyOutPoint` have a specific *meaning* that isn't "this value was not initialized because of runtime details"?

If it does, we should keep it. If it does not, we should discard it in favor of using an `Option` to indicate that it just simply wasn't set.